client/request.go

```diff
@@ -0,0 +1,102 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path"
+
+	"github.com/giantswarm/credentiald/service/creator"
+	"github.com/giantswarm/credentiald/service/lister"
+	"github.com/giantswarm/credentiald/service/searcher"
+	"github.com/giantswarm/microerror"
+)
+
+func (c Client) Create(ctx context.Context, request creator.Request) (*creator.Response, error) {
```

Comment:
What about having separate files like we have with the resource implementations?

Reply:
Your wish is my command. Done